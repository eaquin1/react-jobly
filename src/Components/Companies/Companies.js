import React, {useEffect, useState } from "react"
import JoblyApi from "../../Helpers/JoblyApi";
import CompanyCard from "../Cards/CompanyCard"
function Companies() {
const [companies, setCompanies] = useState();

useEffect(() => {
    async function getCompanies() {

     let c = await JoblyApi.getCompanies();
     setCompanies(c);
    }
    getCompanies()
}, [])

if (!companies) {
    return <div>Loading...</div>
}
    return (
        <div>
        <h1>Companies</h1>
        {companies.map((company) => (
            <CompanyCard item={company} />
        ))}
    </div>
    )
}

export default Companies