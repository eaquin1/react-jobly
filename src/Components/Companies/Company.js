import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import JobCard from "../Cards/JobCard";
import JoblyApi from "../../Helpers/JoblyApi";

function Company() {
    const { handle } = useParams();

    const [company, setCompany] = useState();

    useEffect(() => {
        async function getJobs() {
            let company = await JoblyApi.getCompany(handle);
            
            setCompany(company);
            console.log(company)
        }
        getJobs();
    }, [handle]);

    if (!company) {
        return <div>Loading...</div>;
    }

    return (
        <div className="col-md-8">
            <h1>{company.name}</h1>
            <p>{company.description}</p>
            {company.jobs.map((job) => (
                <JobCard item={job} key={job.id}/>
            ))}
        </div>
    );
}

export default Company;
