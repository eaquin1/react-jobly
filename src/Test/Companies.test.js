import React from 'react';
import { render } from '@testing-library/react';
import Companies from '../Components/Companies/Companies';
import Company from '../Components/Companies/Company';
import { MemoryRouter } from 'react-router';
import { UserProvider } from '../testUtils';

//Testing Companies and Company Details

describe('Testing Companies and Company Details', () => {
    it('it renders Companies without crashing', () => {
      render(
        <MemoryRouter>
          <Companies />
        </MemoryRouter>
      );
    });
    it('it renders CompanyDetails without crashing', () => {
      render(
        <MemoryRouter>
          <UserProvider>
            <Company />
          </UserProvider>
        </MemoryRouter>
      );
    });
    it('matches snapshot', () => {
      const { asFragment } = render(
        <MemoryRouter>
          <Companies />
        </MemoryRouter>
      );
      expect(asFragment()).toMatchSnapshot();
    });
    it('matches snapshot', () => {
      const { asFragment } = render(
        <MemoryRouter>
          <UserProvider>
            <Company />
          </UserProvider>
        </MemoryRouter>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });