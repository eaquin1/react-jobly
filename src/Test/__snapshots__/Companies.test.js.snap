// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Companies and Company Details matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Company not found. Please return back to
    <a
      href="/companies"
    >
       Companies
    </a>
  </div>
</DocumentFragment>
`;

exports[`Testing Companies and Company Details matches snapshot 2`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
